src/libraries/System.Private.CoreLib/src/System/OperatingSystem.cs

```diff
@@ -331,17 +331,16 @@ private static bool IsOSVersionAtLeast(int major, int minor, int build, int revi
             {
                 return current.Major > major;
             }
-            if (current.Minor != minor)
+            if (current.Minor >= 0 && current.Minor != minor)
             {
                 return current.Minor > minor;
             }
-            if (current.Build != build)
+            if (current.Build >= 0 && current.Build != build)
             {
                 return current.Build > build;
             }
 
-            return current.Revision >= revision
-                || (current.Revision == -1 && revision == 0); // it is unavailable on OSX and Environment.OSVersion.Version.Revision returns -1
+            return current.Revision < 0 || current.Revision >= revision;
```

Comment:
```suggestion
            if (current.Revision < 0)
            {
                // Unspecified component satisfies any required version
                return true;
            }
            return current.Revision >= revision;
```
It may be nice to write it like this to make it clear that Build and Revision are handled the same way.